MyBooking/Views/BookingView/src/App.js

```diff
@@ -1,35 +1,75 @@
 import React from 'react';
-import {Container, Row, Col} from 'react-bootstrap';
+import { Container, Row, Col, Modal } from 'react-bootstrap';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import './App.css';
 import FilerAdverts from './components/FilterAdverts';
 import Adverts from './containers/Adverts';
 import Header from './containers/HeaderContainer';
+import SignUp from './components/SignUp';
+import SignIn from './components/SignIn';
+import AddAdvert from './components/AddAdvert';
 
 class App extends React.Component {
     state = {
         searchResult: [],
         adverts: [],
         selectedAddress: '',
-        user: null
+        user: {
+            role: null,
+            name: null,
+            email: null
+        },
+        isSignUpModalOpen: false,
+        isSignInModalOpen: false,
+        logInErrorMessage: null,
+        addAdvretErrorMessage: null,
+        isAddAdvertModalOpen: false
     };
 
     componentDidMount() {
         // fetch method to retrieve user data fro server and set to role
-        const user = this.fetchUserData();
-        this.setState({ user });
-        this.state.searchResult = this.getAdverts();
+        this.setState({
+            searchResult: this.getAdverts(),
+            adverts: this.getAdverts()
+        });
     }
 
-    fetchUserData = () => {
-        // get user data from request
-        return {
-            role: 'unauthorized',
+    handleRegisterUser = params => {
+        //TODO: Add axios fetcher to save user data
+
+        //On success
+        const user = {
+            role: 'landlord',
             name: 'test',
             email: 'test@test.com'
-        }
+        };
+        this.setState({ user });
+        //On error
     };
 
+    handleLogInUser = params => {
+        //TODO: Add axios fetcher to save user data
+
+        //On success
+        const user = {
+            role: 'tenant',
+            name: 'test',
+            email: 'test@test.com'
+        };
+        this.setState({ user, isSignInModalOpen: false });
+        //On error
+        // this.setState( { logInErrorMessage: 'No such user or password is incorrect' } )
+    };
+
+    handleAddAdvert = params => {
+        //TODO: Add axios call
+
+        //On success
+        // this.setState({isAddAdvertModalOpen: false});
+        //On error
+        // this.setState( { addAdvretErrorMessage: 'Please fill all fields' } )
+    }
```

Comment:
Is this a note or documentation?
I think we need to fix a typo `{addAdvretErrorMessage: `

Reply:
This is a reminder where I need to use axions and how I need to check responses

Reply:
where do you see a typo? if you are about the message, it will be fixed, it depends on which status code I can get from the server

Reply:
I'm talking about a word `Advret`.

Reply:
This will be fixed in this or next?

Reply:
oh, I will fix it now